<launch>
  <param name="use_sim_time" value="true"/>

  <node if="1" name="player" pkg="rosbag" type="play" output="screen"
        args="-r 1.0 -s 5 -q --clock /home/victorr/data/anymal/mission_2019-06-11-15-47-14_voxgraph_input.bag"/>

  <node pkg="tf" type="static_transform_publisher" name="odom_to_mission" args="0 0 0 0 0 0 1 odom mission 100"/>

  <node name="voxgraph_mapper" pkg="voxgraph" type="voxgraph_mapping_node" output="screen" args="-alsologtostderr">
<!--          launch-prefix="gdb -ex run &#45;&#45;args">-->
    <rosparam>
      verbose: false
      debug: true
      submap_creation_interval: 10
      pointcloud_topic: "/point_cloud_filter/velodyne/point_cloud_cropped"
      subscriber_queue_length: 10
      input_odom_frame: "odom"
      input_base_link_frame: "base"
      output_mission_frame: "mission"
      output_base_link_frame: "imu_voxgraph"
      output_sensor_frame: "sensor_corrected"
      tsdf_integrator:
        tsdf_voxel_size: 0.20
        truncation_distance: 0.60
        max_ray_length_m: 16.0
        use_const_weight: true
        use_weight_dropoff: true
        use_sparsity_compensation_factor: true
        sparsity_compensation_factor: 20.0
        integration_order_mode: "sorted"
      measurements:
        submap_registration:
          enabled: true
          sampling_ratio: 0.05
          registration_method: "explicit_to_implicit"
          information_matrix:
            x_x:     1.0
            y_y:     1.0
            z_z:     1.0
            yaw_yaw: 1.0
        odometry:
          enabled: true
          information_matrix:
            x_x:     1.0
            y_y:     1.0
            z_z:     2500.0
            yaw_yaw: 2500.0
        height:
          enabled: false
          information_zz: 2500.0
    </rosparam>
  </node>
</launch>
